pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -152,42 +165,292 @@ func (p *k8sPlugin) Attest(ctx context.Context, req *workloadattestor.AttestRequ
-		if !notAllContainersReady || attempt >= p.maxPollAttempts {
-			log.Printf("container id %q not found (attempt %d of %d)", containerID, attempt, p.maxPollAttempts)
+		if !notAllContainersReady || attempt >= config.MaxPollAttempts {
+			log.Printf("container id %q not found (attempt %d of %d)", containerID, attempt, config.MaxPollAttempts)
 			return nil, k8sErr.New("no selectors found")
 		}
 
 		// wait a bit for containers to initialize before trying again.
-		log.Printf("container id %q not found (attempt %d of %d); trying again in %s", containerID, attempt, p.maxPollAttempts, p.pollRetryInterval)
+		log.Printf("container id %q not found (attempt %d of %d); trying again in %s", containerID, attempt, config.MaxPollAttempts, config.PollRetryInterval)
 
 		select {
-		case <-time.After(p.pollRetryInterval):
+		case <-p.clock.After(config.PollRetryInterval):
 		case <-ctx.Done():
 			return nil, k8sErr.New("no selectors found: %v", ctx.Err())
 		}
 	}
 }
 
-func (p *k8sPlugin) getPodListFromInsecureKubeletPort() (out *podList, err error) {
-	httpResp, err := p.httpClient.Get(fmt.Sprintf("http://localhost:%d/pods", p.kubeletReadOnlyPort))
+func (p *k8sPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (resp *spi.ConfigureResponse, err error) {
+	// Parse HCL config payload into config struct
+	config := new(k8sHCLConfig)
+	if err := hcl.Decode(config, req.Configuration); err != nil {
+		return nil, k8sErr.New("unable to decode configuration: %v", err)
+	}
+
+	// Determine max poll attempts with default
+	maxPollAttempts := config.MaxPollAttempts
+	if maxPollAttempts <= 0 {
+		maxPollAttempts = defaultMaxPollAttempts
+	}
+
+	// Determine poll retry interval with default
+	var pollRetryInterval time.Duration
+	if config.PollRetryInterval != "" {
+		pollRetryInterval, err = time.ParseDuration(config.PollRetryInterval)
+		if err != nil {
+			return nil, k8sErr.New("unable to parse poll retry interval: %v", err)
+		}
+	}
+	if pollRetryInterval <= 0 {
+		pollRetryInterval = defaultPollRetryInterval
+	}
+
+	// Determine which kubelet port to hit. Default to the secure port if none
+	// is specified (this is backwards compatible because the read-only-port
+	// config value has always been required, so it should already be set in
+	// existing configurations that rely on it).
+	if config.KubeletSecurePort > 0 && config.KubeletReadOnlyPort > 0 {
+		return nil, k8sErr.New("cannot use both the read-only and secure port")
+	}
+	kubeletPort := config.KubeletReadOnlyPort
+	scheme := "http"
+	if kubeletPort <= 0 {
+		kubeletPort = config.KubeletSecurePort
+		scheme = "https"
+	}
+	if kubeletPort <= 0 {
+		kubeletPort = defaultSecureKubeletPort
+		scheme = "https"
+	}
+
+	// Determine the node name
+	nodeName := p.getNodeName(config.NodeName, config.NodeNameEnv)
+
+	// Configure the HTTP client
+	var token string
+	var tlsConfig *tls.Config
+	if scheme == "https" {
+		// Formulate the kubelet URL.
```

Comment:
s/URL/TLS config/ ?

Reply:
Nit: this whole section might do well to be tucked away in a helper like
```
token, tlsConfig, err := getClientCreds(config)
```

Reply:
Is it possible for these credentials to rotate? Should we be loading them more often?

Reply:
I'll have it opportunistically reload every minute. If it can't reload for whatever reason, I'll keep the old config.